.github/actions/gradle/download/action.yml

```diff
@@ -11,22 +11,25 @@ runs:
   steps:
     - name: Download latest Gradle build validation scripts
       run: |
-        authHeader=""
+        # Set curl general options
+        curlOpts="--silent --fail --show-error --retry 3 --retry-delay 5 --location"
+
+        # Set authorization header if needed
         if [ ! -z "${{ inputs.token }}" ]; then
-          authHeader="--header 'Authorization: Bearer ${{ inputs.token }}'"
+          curlOpts="$curlOpts --header 'Authorization: Bearer ${{ inputs.token }}'"
         fi
 
-        # Build the command to get the details of the latest version of the build validation scripts
-        cmdGetLatestReleaseData="curl -s $authHeader https://api.github.com/repos/gradle/gradle-enterprise-build-validation-scripts/releases/latest"
+        # Set curl params to get latest release data of the build validation scripts
+        curlParamsGetLatestReleaseData="$curlOpts https://api.github.com/repos/gradle/gradle-enterprise-build-validation-scripts/releases/latest"
 
-        # Get the download url of the latest version of the build validation scripts
-        downloadUrl=$($cmdGetLatestReleaseData | jq -c '.assets[] | select(.content_type == "application/zip")' | jq -r .browser_download_url | grep -v maven)
+        # Get the download url of the latest release
+        downloadUrl=$(eval curl $curlParamsGetLatestReleaseData | jq -c '.assets[] | select(.content_type == "application/zip")' | jq -r .browser_download_url | grep -v maven)
```

Comment:
Is `eval` needed? Same question for line 32.

Reply:
This is actually required to keep the quotes around the `Authorization` header. 
I tried to use a quote protection approach which would have been better (`\'` or `\"`) but it resulted unsuccessful as the quote was lost at execution time.
Feel free to suggest an alternative solution 🥹

Reply:
Ah, that makes sense! What I would recommend is storing each argument in an array and then expanding the array when `curl` is called. 

We do this _a lot_ throughout the build validation scripts code. For example, see this [`invoke_gradle`](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/b6e9e6d32afba7e41f4491dba2e787f97ea4ebeb/components/scripts/lib/gradle.sh#L46) function where we build out the arguments for calling Gradle.

Reply:
alright, I considered this approach but thought it could create issues on Windows runner, not sure `eval` would behave better though 😅
If this is used inside the scripts, I can definitely use it in the composite action, thanks for the hint, will share something shortly.

Reply:
Ah, I have no idea if it would cause any issues with Windows runners. I see we specify `shell: bash` but I don't know what that means when running on Windows.

Reply:
The Windows runners us WSL (Windows Subsystem for Linux) and execute commands using a full and real version of bash. So using arrays should be fine for the Windows runners as well.

Reply:
just confirming as I successfully tested the action on `windows-runner` 😅